client/battle/CBattleInterface.h

```diff
@@ -121,14 +121,26 @@ class CBattleInterface : public CIntObject
 private:
 	SDL_Surface *background, *menu, *amountNormal, *amountNegative, *amountPositive, *amountEffNeutral, *cellBorders, *backgroundWithHexes;
 
-	CButton *bOptions, *bSurrender, *bFlee, *bAutofight, *bSpell,
-		* bWait, *bDefence, *bConsoleUp, *bConsoleDown, *btactNext, *btactEnd;
-	CBattleConsole *console;
-	CBattleHero *attackingHero, *defendingHero; //fighting heroes
-	CStackQueue *queue;
+	std::shared_ptr<CButton> bOptions;
+	std::shared_ptr<CButton> bSurrender;

```

Comment:
these buttons, shouldn't be unique_ptr?

Reply:
@FeniksFire why?

Usage of ```unique_ptr``` apply certain requirements on how headers are organized too and ours aren't very good. You know...

Reply:
@ArseniyShestakov and why you need shared_ptr? From first point of view something like "button" shouldn't be shared, and even modified. So sole owner for buttons sounds good.

Reply:
It does make sense, but I suppose at least for now we slap ```shared``` on everything so we can keep internals of GH working until we actually rewrite the whole thing. After all GH itself not yet use smart pointers I suppose.

Reply:
All widgets must be `shared_ptr`, later child links in CIntObjects to be changed to shared_ptr and parent link to weak_ptr

Reply:
@FeniksFire  What is incorrectly written?